app/services/toolbars/projects_service.rb

```diff
@@ -13,6 +13,7 @@ def initialize(current_user, project_ids: [], project_folder_ids: [])
       @project_folders = current_user.current_team.project_folders.where(id: project_folder_ids)
 
       @items = @projects + @project_folders
+      @not_assigned = @projects.any? { |project| !project.assigned_users.include?(current_user) }
```

Comment:
This should be done as a single query with a join, since this loop will run a bunch of queries unnecessarily, something like:
```
current_user.current_team.projects.joins(
  "JOIN user_assignments ON user_assignments.assignable_type = 'Project' " \
  "AND user_assignments.assignable_id = projects.id " \
  "AND user_assignments.user_id = #{current_user.id}"
).exists?(user_assignments: { id: nil })
```

Not sure if there's a prettier or a more readable way to do it (maybe with a subquery?).

Reply:
Also maybe extract this to a scope on the Project model

app/models/project.rb:
```
  scope :not_assigned_to_user, lambda(user) {
    joins(
      "LEFT JOIN user_assignments ON user_assignments.assignable_type = 'Project' " \
      "AND user_assignments.assignable_id = projects.id " \
      "AND user_assignments.user_id = #{user.id}"
    ).exists?(user_assignments: { id: nil })
  }
```